SYCL/ESIMD/api/functional/ctors/ctor_default.cpp

```diff
@@ -84,34 +82,49 @@ struct const_ref {
   }
 };
 
-// Functor with the main test routine to iterate over the pre-determined
-// datatypes
-template <typename DataT, int NumElems, typename TestCaseT> struct test {
-  bool operator()(sycl::queue &queue, const std::string &data_type) {
+// Function that verifies obtained test result.
+template <typename DataT, int NumElems, typename TestCaseT>
+bool verify_test_result(const shared_vector<DataT> &result_data,
+                        const std::string &data_type) {
+  bool passed{true};
+  DataT default_val{};
+
+  for (size_t i = 1; i < result_data.size(); ++i) {
+    if (result_data[i] != default_val) {
+      passed = false;
+      log::fail<NumElems>(
+          "In simd by " + TestCaseT::get_description() +
+              " elem value is not equal to default value, retrieved: " +
+              std::to_string(result_data[i]) +
+              ", at index: " + std::to_string(i),
+          data_type);
+    }
+  }
+
+  return passed;
+}
+
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+              const std::string &data_type) {
     bool passed{true};
 
     DataT default_val{};
     using AllocatorT = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
 
-    std::vector<DataT, AllocatorT> result{NumElems, AllocatorT{queue}};
+    shared_vector<DataT> result{NumElems, AllocatorT{queue}};
     queue.submit([&](sycl::handler &cgh) {
       auto out = result.data();
       cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
           [=]() SYCL_ESIMD_KERNEL {
             TestCaseT::template call_simd_ctor<DataT, NumElems>(out);
           });
     });
-    for (const auto &it : result) {
-      if (it != default_val) {
-        passed = false;
-        log::fail<NumElems>(
-            "In simd by " + TestCaseT::get_description() +
-                " elem value is not equal to default value, retrieved: " +
-                std::to_string(it) +
-                ", expected: " + std::to_string(default_val),
-            data_type);
-      }
-    }
+
+    passed = verify_test_result<DataT, NumElems, TestCaseT>(result, data_type);
```

Comment:
You are providing a custom `Verifier` and a custom `verify_test_result` in the same source file. What's the intent behind moving a part of this function into the separate one within the same source file?

Reply:
`verify_test_result` was removed and verification was moved into `Verifier` struct
removing was [here](https://github.com/intel/llvm-test-suite/pull/614/commits/e44cd0f653ddd3f6c73aa85230716eb71cad054e#diff-6d44d6893e30c56cec1c2e698ea43849081d8b092f7a87532bd567294d14ed69L85-L106) emplacing was [here](https://github.com/intel/llvm-test-suite/pull/614/commits/e44cd0f653ddd3f6c73aa85230716eb71cad054e#diff-6d44d6893e30c56cec1c2e698ea43849081d8b092f7a87532bd567294d14ed69R103-R110)